frontend/src/pages/ViewDietPlanPage.tsx

```diff
@@ -141,8 +142,8 @@ export const ViewDietPlanPage = () => {
           ))}
         </SelectContent>
       </Select>
-      <DietPlanForm dietPlan={plan} updateDietPlan={updateDietPlan} />
-      {plan.meals.length > 0 && (
+      {plan && <DietPlanForm dietPlan={plan} updateDietPlan={updateDietPlan} />}
+      {plan && plan.meals.length > 0 && (
```

Comment:
Maybe put them inside the same brackets. You are already doing {plans && ...code} then you do it again for the length check